xc/priority_auths.lua

```diff
@@ -8,40 +8,49 @@ local _M = { }
 -- @return true if authorized, false if denied, nil if unknown
 -- @return reason why the authorization is denied (optional, required only when denied)
 function _M.authorize(service_id, credentials, metric)
-  local redis_pub, ok_pub = redis_pool.acquire()
+  local redis_pub, ok_pub, err = redis_pool.acquire()
 
   if not ok_pub then
+    ngx.log(ngx.WARN, "[priority auths] couldn't connect to redis pub: ", err)
```

Comment:
To me it looks like overcomplicating the things a bit.
Also, for troubleshooting, I actually find it useful to just take the error message found in the logs, and search for where it appears in the code - makes it easy to trace the issue.